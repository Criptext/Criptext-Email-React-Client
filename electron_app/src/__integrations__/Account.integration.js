/* eslint-env node, jest */

const DBManager = require('../database');

const account = {
  recipientId: 'user',
  deviceId: 1,
  name: 'User One',
  registrationId: 2,
  privKey: 'ghi',
  pubKey: 'jkl'
};

const insertAccount = async () => {
  await DBManager.createAccount(account);
};

beforeAll(async () => {
  await DBManager.deleteDatabase();
  await DBManager.initDatabaseEncrypted({
    key: '1111',
    shouldAddSystemLabels: true
  });
  await insertAccount();
});

describe('Update data account from Account Table:', () => {
  it('should update account', async () => {
    const account = {
      jwt: 'abc',
      refreshToken: 'def',
      recipientId: 'user'
    };
    const [isUpdated] = await DBManager.updateAccount(account);
    expect(isUpdated).toBe(1);
    const accounts = await DBManager.getAccount();
    expect(accounts).toMatchSnapshot();
  });
});

describe('Load account contact from Account Table:', () => {
  it('should load account by recipientId', async () => {
    const [account] = await DBManager.getAccountByParams({
      recipientId: 'user'
    });
    expect(account).toMatchObject(
      expect.objectContaining({
        recipientId: 'user'
      })
    );
  });

  it('should load account by deviceId', async () => {
    const [account] = await DBManager.getAccountByParams({ deviceId: '' });
    expect(account).toBeUndefined;
  });
});
