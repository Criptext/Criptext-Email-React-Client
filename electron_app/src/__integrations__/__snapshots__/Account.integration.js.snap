// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update data account from Account Table: should update account 1`] = `
Array [
  Object {
    "deviceId": 1,
    "encryptToExternals": false,
    "id": 1,
    "jwt": "abc",
    "name": "User One",
    "privKey": "ghi",
    "pubKey": "jkl",
    "recipientId": "user",
    "refreshToken": "def",
    "registrationId": 2,
    "signFooter": true,
    "signature": "",
    "signatureEnabled": false,
  },
]
`;
